File: CoreJava/Assignment/src/com/techlabs/assignment/december/ResumeBuilder.java
package com.techlabs.assignment.december;

import java.awt.Desktop;
import java.io.*;

public class ResumeBuilder {
	private String name;
	private String mobile;
	private String age;
	private String percentage;
	private String address;

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getMobile() {
		return mobile;
	}

	public void setMobile(String mobile) {
		this.mobile = mobile;
	}

	public String getAge() {
		return age;
	}

	public void setAge(String age) {
		this.age = age;
	}

	public String getPercentage() {
		return percentage;
	}

	public void setPercentage(String percentage) {
		this.percentage = percentage;
	}

	public String getAddress() {
		return address;
	}

	@NeedRefactoring(str = "This method needs refactoring")
	public void setAddress(String address) {
		this.address = address;
	}

	@NeedRefactoring(str = "This method needs refactoring")
	public void buildResume() {
		try {
			File file = new File("Resumes/resume.html");
			FileReader fr = new FileReader(file);
			BufferedReader br = new BufferedReader(fr);
			String iteration;
			String htmldata = null;
			while ((iteration = br.readLine()) != null) {
				htmldata += iteration;
			}
			br.close();
			htmldata = htmldata.replaceAll("null", "");
			htmldata = htmldata.replaceAll("##name##", getName());
			htmldata = htmldata.replaceAll("##age##", getAge());
			htmldata = htmldata.replaceAll("##address##", getAddress());
			htmldata = htmldata.replaceAll("##percentage##", getPercentage());
			htmldata = htmldata.replaceAll("##mobilenumber##", getMobile());
			FileWriter fw = new FileWriter("Resumes/" + getName()
					+ "Resume.html");
			fw.write(htmldata);
			fw.close();
			File resumefile = new File("Resumes/" + getName() + "Resume.html");
			Desktop.getDesktop().browse(resumefile.toURI());
		} catch (IOException e) {
			e.printStackTrace();
		}
	}
}

File: DotNET/C#/ResumeBuilderApp/ResumeBuilderApp/ResumeBuilder.cs
using System;
using System.IO;
namespace ResumeBuilderApp
{
    class ResumeBuilder
    {
        private String _name;
        private String _mobile;
        private String _age;
        private String _percentage;
        private String _address;

        public ResumeBuilder(String name, String mobile, String age, String percentage, String address)
        {
            _name = name;
            _mobile = mobile;
            _age = age;
            _percentage = percentage;
            _address = address;
        }

        public String Name
        {
            get
            {
                return _name;
            }
        }

        public String Mobile
        {
            get
            {
                return _mobile;
            }
        }

        public String Age
        {
            get
            {
                return _age;
            }
        }

        public String Percentage
        {
            get
            {
                return _percentage;
            }
        }

        public String Address
        {
            get
            {
                return _address;
            }
        }

        public void BuildResume()
        {
            StreamReader streamreader = new StreamReader("Resumes/resume.html");
            String htmldata = "";
            using (streamreader)
            {
                while (streamreader.ReadLine() != null)
                {
                    htmldata += streamreader.ReadLine();
                }
                htmldata = htmldata.Replace("null", "");
                htmldata = htmldata.Replace("##name##", Name);
                htmldata = htmldata.Replace("##age##", Age);
                htmldata = htmldata.Replace("##address##", Address);
                htmldata = htmldata.Replace("##percentage##", Percentage);
                htmldata = htmldata.Replace("##mobilenumber##", Mobile);
            }
            String fileName = "Resumes/" + Name + "Resume.html";
            File.WriteAllText(fileName, htmldata);
            Console.WriteLine(fileName);
        }

    }
}
